tests/robustness/coverage/key_pattern_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2025 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coverage_test
+
+import (
+	"strings"
+	"testing"
+)
+
+func pattern(k string) (string, bool) {
+	if k == "/registry/health" || k == "compact_rev_key" {
+		return k, true
+	}
+	k, valid := strings.CutPrefix(k, "/registry")
+	if !valid {
+		return k, false
+	}
+	slash := ""
+	if strings.HasSuffix(k, "/") {
+		slash = "/"
+	}
+	s := strings.Split(strings.Trim(k, "/"), "/")
+
+	resource, shift := resourceOrCRD(s[0])
+	s = s[shift:]
+
+	if len(s) == 0 {
+		// Listing all resources.
+		return "/registry/" + resource + slash, true
+	}
+	if len(s) == 1 && slash == "" {
+		// Get on a non-namespaced resource.
+		return "/registry/" + resource + "/{name}", true
+	}
+	if len(s) == 1 {
+		// Listing all resources in a namespace.
+		return "/registry/" + resource + "/{namespace}" + slash, true
+	}
+	if len(s) == 2 && slash == "" {
+		// Get on a namespaced resource.
+		return "/registry/" + resource + "/{namespace}/{name}", true
+	}
+
+	return k, false
+}
+
+func resourceOrCRD(s string) (string, int) {
+	if strings.Contains(s, ".") {
+		// CRD
+		return "{api-group}/{resource}", 2
+	}
+	switch s {
+	case "services":
+		// services have subresources specs and endpoints.
+		return "{resource}", 2
+	case "masterleases":
+		// masterleases are a special case as they are not a regular (namespaced) resource.
+		return "masterleases", 1
+	case "events":
+		// events are a special case as they have watch cache disabled by default.
+		return "events", 1
+	}
+	return "{resource}", 1
+}
+
+func TestPatternValid(t *testing.T) {
+	for _, tc := range []struct {
+		name string
+		key  string
+		want string
+	}{
+		{
+			name: "special_root_key",
+			key:  "compact_rev_key",
+			want: "compact_rev_key",
+		},
+		{
+			name: "special_healthcheck",
+			key:  "/registry/health",
+			want: "/registry/health",
+		},
+		{
+			name: "core_namespace_get",
```

Comment:
Why call it "core_namespace_get"  and not "core_namespace_name" ?

Also `core` is K8s concept that might not be known for etcd users. Maybe call it `coreAPI_namespace_name`